### PR TITLE
ICU-21900 Update ICU4J Change report (BRS#20)

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/APIChangeReport.html [hunk 1]
```diff
@@ -2,233 +2,171 @@
...
-<li><span style='color:green'>(stable)</span> public static boolean <i>isTrailSurrogate</i>(char)</li>
-</ul>
-</ul>
-
+<h2>Removed from ICU4J 70.1</h2>
+<p>(no API removed)</p>
 
 <hr/>
-<h2>Deprecated or Obsoleted in ICU4J 70.1</h2>
+<h2>Deprecated or Obsoleted in ICU4J 71.0.1</h2>
```
Comment (yumaoka) [inline icu4j/APIChangeReport.html:17]: I'm expecting 71.1 at this point. We need ICU4J version updated to 71.1 before generating the change report.

**Review by yumaoka — CHANGES_REQUESTED** [review]
We need ICU version number change done first.

File: icu4j/APIChangeReport.html (hunk 1 above)
Comment (sven-oly) [reply icu4j/APIChangeReport.html:17]: Would it be sufficient to have ICU4J 71 without any minor version info?

Comment (markusicu) [reply icu4j/APIChangeReport.html:17]: Yes, that's what we said today. Just 70 vs. 71.

**Comment by sven-oly** [comment]
PTAL

File: icu4j/APIChangeReport.html (hunk 1 above)
Comment (sven-oly) [reply icu4j/APIChangeReport.html]: This should be fixed now.

**Review by markusicu — APPROVED** [review]
lgtm

**Pull request merged**